### DISCUSSION TITLE
Rabbit server fails to start with `credentials_obfuscation` error

### DISCUSSION THREAD

**Discussion by Zorato** [rabbitmq/rabbitmq-server#2881] [Other] [answered]
Hello, after installing latest `rabbit-server` (3.8.14) on Ubuntu 18.04 it fails to start with 
```
{"Kernel pid terminated",application_controller,"{application_start_failure,credentials_obfuscation,{{shutdown,{failed_to_start_child,credentials_obfuscaton_svc,{undef,[{crypto,cipher_info,[aes_128_cbc],[]},{credentials_obfuscation_pbe,iv_length,1,[{file,\"src/credentials_obfuscation_pbe.erl\"},{line,125}]},{credentials_obfuscation_pbe,encrypt,5,[{file,\"src/credentials_obfuscation_pbe.erl\"},{line,75}]},{credentials_obfuscation_svc,check,3,[{file,\"src/credentials_obfuscation_svc.erl\"},{line,163}]},{credentials_obfuscation_svc,init_state,0,[{file,\"src/credentials_obfuscation_svc.erl\"},{line,133}]},{gen_server,init_it,2,[{file,\"gen_server.erl\"},{line,374}]},{gen_server,init_it,6,[{file,\"gen_server.erl\"},{line,342}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,249}]}]}}},{credentials_obfuscation_app,start,[normal,[]]}}}"}
Kernel pid terminated (application_controller) ({application_start_failure,credentials_obfuscation,{{shutdown,{failed_to_start_child,credentials_obfuscaton_svc,{undef,[{crypto,cipher_info,[aes_128_cbc

Crash dump is being written to: erl_crash.dump...done
```

is there any clue how to fix this?

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment, marked as answer]
```
{undef,[{crypto,cipher_info,[aes_128_cbc],[]}
```

is the line you are looking for. `crypto:cipher_info/1` is a function that's not available in your Erlang installation. Either it does not include crypto/TLS modules at all (e.g. OpenSSL was not found when building) or you are running an Erlang version that is too old.

Ubuntu 18.04 provides Erlang 20.2 and [RabbitMQ 3.8.14 requires 22.3 or 23.x](https://github.com/rabbitmq/rabbitmq-server/releases/tag/v3.8.14). See [Installing Erlang/OTP on Debian and Ubuntu](https://www.rabbitmq.com/which-erlang.html#debian).

**Reply by michaelklishin** [reply]
Which leads to [Where to Get Recent Erlang Version on Debian and Ubuntu](https://www.rabbitmq.com/install-debian.html#erlang-repositories).

**Reply by Zorato** [reply]
Thank you! Installing latest Erlang version helped.

**Reply by JamesYLiberis** [reply]
Any one came across this problem?

Rabbitmq:3.8.14

when run:

$ sudo systemctl start rabbitmq-server

got this credentials obfuscation problem:

{"Kernel pid terminated",application_controller,"{application_start_failure,cr A nel pid terminated",application_controller,"{application_start_failure,credentials_obfuscation,{{shutdown,{failed_to_start_child,credentials_obfu l redentials_obfuscation,{{shutdown,{failed_to_start_child,credentials_obfuscaton_svc,{undef,[{crypto,cipher_info,[aes_128_cbc],[]},{credentials_ob ede uscaton_svc,{undef,[{crypto,cipher_info,[aes_128_cbc],[]},{credentials_obfuscation_pbe,iv_length,1,[{file,"src/credentials_obfuscation_pbe.erl" sca bfuscation_pbe,iv_length,1,[{file,"src/credentials_obfuscation_pbe.erl"},{line,125}]},{credentials_obfuscation_pbe,encrypt,5,[{file,"src/crede "},{line,125}]},{credentials_obfuscation_pbe,encrypt,5,[{file,"src/credentials_obfuscation_pbe.erl"},{line,75}]},{credentials_obfuscation_svc,c entials_obfuscation_pbe.erl"},{line,75}]},{credentials_obfuscation_svc,check,3,[{file,"src/credentials_obfuscation_svc.erl"},{line,163}]},{cre check,3,[{file,"src/credentials_obfuscation_svc.erl"},{line,163}]},{credentials_obfuscation_svc,init_state,0,[{file,"src/credentials_obfuscati edentials_obfuscation_svc,init_state,0,[{file,"src/credentials_obfuscation_svc.erl"},{line,133}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,249}]}]}}},{credentials_obfuscation_app,start,[normal,[]]}}}"}

**Comment by ArcticSnowman** [comment]
```
{"Kernel pid terminated",application_controller,"{application_start_failure,credentials_obfuscation,{{shutdown,{failed_to_start_child,credentials_obfuscaton_svc,{low_entropy,[{crypto,strong_rand_bytes,1,[{file,\"crypto.erl\"},{line,1141}]},{credentials_obfuscation_svc,check,3,[{file,\"src/credentials_obfuscation_svc.erl\"},{line,197}]},{credentials_obfuscation_svc,init_state,0,[{file,\"src/credentials_obfuscation_svc.erl\"},{line,169}]},{gen_server,init_it,2,[{file,\"ge
n_server.erl\"},{line,851}]},{gen_server,init_it,6,[{file,\"gen_server.erl\"},{line,814}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,240}]}]}}},{credentials_obfuscation_app,start,[normal,[]]}}}"}
Kernel pid terminated (application_controller) ({application_start_failure,credentials_obfuscation,{{shutdown,{failed_to_start_child,credentials_obfuscaton_svc,{low_entropy,[{crypto,strong_rand_bytes,1,[{file,"crypto.erl"},{line,1141}]},{credentials_obfuscation_svc,check,3,[{file,"src/credentials_obfuscation_svc.erl"},{line,197}]},{credentials_obfuscation_svc,init_state,0,[{file,"src/credentials_obfuscation_svc.erl"},{line,169}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,851}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,814}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}}},{credentials_obfuscation_app,start,[normal,[]]}}})
``` 

low_entropy?     Any idea?

Ubuntu 20.04
erlang-crypto/focal,now 1:25.3-1 amd64 [installed,automatic]

**Reply by lukebakken** [reply]
@ArcticSnowman Hello! Just FYI you probably should have started a new discussion for this error because it is not the same as the original here.

At any rate, `low_entropy` is mentioned here a few times - https://www.erlang.org/doc/man/crypto.html

https://stackoverflow.com/questions/30650842/how-to-handle-low-entropy-exception-of-cryptostrong-rand-bytesn

What hardware or virtual machine layer are you using in your environment?

**Reply by ArcticSnowman** [reply]
I have since found that the issue is due to FIPS requirement and that the default ERLANG packages do not has FIPS enables.  Thanks
